File: nj2k/testData/newJ2k/constructors/genericIdentifier.java
public class Identifier<T> {
  private final T myName;
  private boolean myHasDollar;
  private boolean myNullable = true;

  public Identifier(T name) {
    myName = name;
  }

  public Identifier(T name, boolean isNullable) {
    myName = name;
    myNullable = isNullable;
  }

  public Identifier(T name, boolean hasDollar, boolean isNullable) {
    myName = name;
    myHasDollar = hasDollar;
    myNullable = isNullable;
  }

  @Override
  public T getName() {
    return myName;
  }
}

public class User {
  public static void main() {
     Identifier<?> i1 = new Identifier<String>("name", false, true);
     Identifier i2 = new Identifier<String>("name", false);
     Identifier i3 = new Identifier<String>("name");
  }
}

File: j2k/tests/testData/ast/constructors/genericIdentifier.kt
public fun <T> Identifier(name: T): Identifier<T> {
    return Identifier(name, false)
}

public fun <T> Identifier(name: T, isNullable: Boolean): Identifier<T> {
    val __ = Identifier(name, false)
    __.myNullable = isNullable
    return __
}

public fun <T> Identifier(name: T, hasDollar: Boolean, isNullable: Boolean): Identifier<T> {
    val __ = Identifier(name, hasDollar)
    __.myNullable = isNullable
    return __
}

public class Identifier<T>(private val myName: T, private val myHasDollar: Boolean) {
    private var myNullable = true

    public fun getName(): T {
        return myName
    }
}

public class User {
    class object {
        public fun main() {
            val i1 = Identifier("name", false, true)
            val i2 = Identifier<String>("name", false)
            val i3 = Identifier("name")
        }
    }
}